client/karma.js

```diff
@@ -128,8 +128,10 @@ function Karma (updater, socket, iframe, opener, navigator, location, document)
       }
     // run in iframe
     } else {
+      iframe.onload = () => {
+        karmaNavigating = false
```

Comment:
This change would delay the check on unload events until the test was completely loaded.  Thus code that executes during the load sequence and navigates the page will be allowed to execute, contrary to our goal of preventing navigation during testing.

Rather than a race is it possible that your test is attempting to navigate during test load?